tests/memgraph/test_query_builder.py

```diff
@@ -593,7 +593,61 @@ def test_add_string_complete(memgraph):
     mock.assert_called_with(expected_query)
 
 
-def test_node_object(memgraph):
+def test_node_instance(memgraph):
+    class User(Node):
+        name: Optional[str] = Field(index=True, unique=True, db=memgraph)
+
+    user = User(name="Ron").save(memgraph)
+    query_builder = QueryBuilder().match().node(node=user, variable="u").return_()
+    expected_query = " MATCH (u:User {name: 'Ron'}) RETURN * "
+
+    with patch.object(Memgraph, "execute_and_fetch", return_value=None) as mock:
+        query_builder.execute()
+
+    mock.assert_called_with(expected_query)
+
+
+def test_unsaved_node_instance(memgraph):
+    class User(Node):
+        name: Optional[str] = Field(index=True, unique=True, db=memgraph)
+
+    user = User(name="Ron")
+    query_builder = QueryBuilder().match().node(node=user, variable="u").return_()
+    expected_query = " MATCH (u:User {name: 'Ron'}) RETURN * "
+
+    with patch.object(Memgraph, "execute_and_fetch", return_value=None) as mock:
+        query_builder.execute()
+
+    mock.assert_called_with(expected_query)
+
+
+def test_node_relationship_instances(memgraph):
+    class User(Node):
```

Comment:
Can we promote these in-test defined classes to fixtures, and use them somehow as parameters for the tests?

Reply:
I'm just guessing, for fixture, don't you need an instance? In that case, just put it somewhere outside the test?
Or was it some issue that we couldn't have a class outside a test? I'm sure it can be somehow non duplicated

Reply:
Yes, we can't have the classes outside of the test method scope. 
As we discussed, we will ignore this problem for this release but are committed to creating a factory that will remove the duplicated code.